src/node_i18n.cc

```diff
@@ -501,19 +500,30 @@ void ConverterObject::Decode(const FunctionCallbackInfo<Value>& args) {
         converter->set_bom_seen(true);
       }
     }
-    ret = ToBufferEndian(env, &result);
-    if (omit_initial_bom && !ret.IsEmpty()) {
+
+    Local<Value> error;
+    char16_t* output = result.out();
```

Comment:
`result` is a `MaybeStackBuffer<UChar>`, so shouldn't this be 
```suggestion
    UChar* output = result.out();
```

or probably even rather 
```suggestion
    const UChar* output = result.out();
```